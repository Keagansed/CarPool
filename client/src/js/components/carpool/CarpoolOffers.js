// File Type: Component

import { observer } from 'mobx-react';
import React, { Component } from 'react';

import CarpoolOffer from './CarpoolOffer';
import OfferStore from '../../stores/OfferStore';

/*
* Purpose: container for the carpool offers that a user has.
*/
@observer class CarpoolOffers extends Component {

    /*
    * Purpose: accesses the store to call the 'getOffers' function which gets and sets the offers
    * of this user.
    */
    componentDidMount() {
        this.props.store.getOffers(this.props.token);
    }

    /*
    * Purpose: iterates through each offer, creating a CarpoolOffer component for each which is
    * then returned unless there are no carpool offers.
    */
    renderOffers() {
        const Offers = this.props.store.offers.map(offer =>             
            <CarpoolOffer key={offer._id} offerId={offer._id} store={new OfferStore(offer.CarpoolName, offer.SenderID, offer.SenderRoute, offer.RecieverID, offer.RecieverRoute, offer.JoinRequest)}/>
        )
        
        if(Offers.length > 0) {
            return Offers;
        }else{
            return(
                <h5 className="txt-center mtop-10px txt-white">
                    No Offers
                </h5>
            );
        }
    }

    /*
    * Purpose: displays a spinner while the carpools are loading.
    */
    renderLoading = () => {
        return(
            <div>
                <div className="spinner">
                    <div className="double-bounce1"></div>
                    <div className="double-bounce2"></div>
                </div>
            </div>
        )
    }

    /*
    * Purpose: renders the component in the DOM.
    */
    render() {

        if(this.props.store.loadingOffers){
            return(
                <div className="scroll-vert">
                    {this.renderLoading()}
                </div>
            );
        }else{
            return(
                <div>
                    {this.renderOffers()}
                </div>
            );
        }
    }
}

export default CarpoolOffers;