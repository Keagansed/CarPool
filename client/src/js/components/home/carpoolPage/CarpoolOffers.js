import { observer } from 'mobx-react';
import React, { Component } from 'react';
import CarpoolOffer from './CarpoolOffer';
import OfferStore from '../../../stores/OfferStore';

@observer class CarpoolOffers extends Component{
    componentDidMount(){
        this.props.store.getOffers(this.props.token);
    }


    renderOffers(){
        const Offers = this.props.store.offers.map(offer =>             
            <CarpoolOffer key={offer._id} store={new OfferStore(offer.CarpoolName, offer.SenderID, offer.SenderRoute, offer.RecieverID, offer.RecieverRoute, offer.JoinRequest)}/>
        )
        
        if(Offers.length > 0) {
            return Offers;
        }else {
            return(
                <h5 className="txt-center mtop-50px txt-white">
                    No Offers{console.log(this.props.token)}
                </h5>
            );
        }
    }

    renderLoading = () => {
        return(
            <div>
                <div className="spinner">
                    <div className="double-bounce1"></div>
                    <div className="double-bounce2"></div>
                </div>
            </div>
        )
    }

    render(){
        if (this.props.store.loadingOffers){
            return(
                <div className="scroll-vert">
                    {this.renderLoading()}
                </div>
            );
        }
        else{
            return(
                <div>
                    {this.renderOffers()}
                </div>
            );
        }
    }
}

export default CarpoolOffers;