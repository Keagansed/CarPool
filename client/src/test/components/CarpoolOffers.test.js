import React from 'react';
import CarpoolOffers from '../../js/components/carpool/CarpoolOffers';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

class MockStore {
    offers = [];
    createCarpoolOffer = false;
    joinCarpoolOffer = false;

    getOffers = () => {
        this.offers.push({
            CarpoolName : "TestCarpool",
            SenderID : "sender123",
            SenderRoute : {},
            RecieverID : "reciever123",
            RecieverRout : {},
            JoinRequest : false
        });
    };

    makeOffer = (carpoolName, senderId, senderRoute, recieverId, recieverRoute, join, carpoolID) => {
        createCarpoolOffer = true;
    };

    makeOfferToJoin = (carpoolName, senderId, senderRoute, recieverId, recieverRoute, join, carpoolID) => {
        joinCarpoolOffer = true;
    };
}

describe('Carpool Offers Component', () => {
    let token, container, instance, mockStore;

    beforeEach(() => {
        mockStore = new MockStore();
        token = "testtoken123";
        
        container = shallow(<CarpoolOffers store={mockStore} token={token}/>);
        instance = container.instance();
    });

    it ('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><CarpoolOffers store={mockStore} token={token}/></MemoryRouter>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it ('renders correctly', () => {
        expect(container.length).toEqual(1);
    });

    it('populates the offers', () => {
        expect(instance.props.store.offers.length).toEqual(1);
    });

});