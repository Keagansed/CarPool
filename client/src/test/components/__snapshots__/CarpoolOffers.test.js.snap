// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carpool Offers Component captures snapshot 1`] = `
<div>
  <div>
    <div
      className="container-fluid bg-purple bordbot-2px-white"
      onClick={[Function]}
    >
      <div
        className="row txt-white padver-10px"
      >
        <div
          className="col-9"
        >
          <div
            className="col-12"
          >
            <h5>
              TestCarpool
            </h5>
          </div>
          <div
            className="col-12"
          >
            Create a new carpool
          </div>
        </div>
        <div
          className="col-3 vertical-right"
        >
          <div
            className="col-12"
          >
            <h5>
              <i
                className="fa fa-info-circle"
              />
            </h5>
          </div>
          <div
            className="col-12"
          />
        </div>
      </div>
    </div>
    <div
      className="modal"
      id="myModal"
      role="dialog"
      style={
        Object {
          "display": "none",
        }
      }
      tabIndex="-1"
    >
      <div
        className="modal-dialog"
        role="document"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header bg-aqua"
          >
            <h5
              className="modal-title fw-bold"
            >
              TestCarpool
            </h5>
            <button
              aria-label="Close"
              className="close"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
          <div
            className="modal-body"
          >
            <div
              className="row bordbot-1px-dash-grey"
            >
              <h6
                className="fw-bold mx-auto"
              >
                Offer Sent By
              </h6>
            </div>
            <div
              className="row bordbot-1px-dash-grey mbottom-10px"
            >
              <div
                className="col-7 padright-0"
              >
                undefined undefined
              </div>
              <div
                className="col-5 vertical-right"
              >
                <a
                  href="/ProfilePage/sender123"
                  onClick={[Function]}
                >
                  View Profile
                </a>
              </div>
            </div>
            <div
              className="row"
            >
              <p
                className="mx-auto mbottom-0"
              >
                This is an invite to create a new carpool
              </p>
            </div>
            <div
              className="row mtop-10px "
            >
              <button
                className="col-5 btn btn-primary mx-auto width-15rem brad-2rem mbottom-0 bg-green txt-white fw-bold"
                onClick={[Function]}
              >
                Accept
              </button>
              <button
                className="col-5 btn btn-primary mx-auto width-15rem brad-2rem mbottom-0 bg-red txt-white fw-bold"
                onClick={[Function]}
              >
                Decline
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
